.github/workflows/pre-commit.yaml

```
@@ -0,0 +1,38 @@
+name: pre-commit
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
```

```suggestion
        uses: actions/checkout@v4
```

https://github.com/actions/checkout/releases/tag/v4.1.7